ninja/main.py

```diff
@@ -514,6 +519,28 @@ def on_exception(self, request: HttpRequest, exc: Exc[_E]) -> HttpResponse:
             raise exc
         return handler(request, exc)
 
+    def validation_error_from_error_contexts(
+        self, error_contexts: List[ValidationErrorContext]
+    ) -> ValidationError:
+        errors: List[Dict[str, Any]] = []
+        for context in error_contexts:
+            model = context.model
+            e = context.pydantic_validation_error
+            for i in e.errors(include_url=False):
+                i["loc"] = (
+                    model.__ninja_param_source__,
+                ) + model.__ninja_flatten_map_reverse__.get(i["loc"], i["loc"])
+                # removing pydantic hints
+                del i["input"]  # type: ignore
+                if (
+                    "ctx" in i
+                    and "error" in i["ctx"]
+                    and isinstance(i["ctx"]["error"], Exception)
+                ):
+                    i["ctx"]["error"] = str(i["ctx"]["error"])
+                errors.append(dict(i))
+        return ValidationError(errors)
+
```

Comment:
I was thinking of exposing this on Routers or api_operations instead, but since exception handlers must be registered on APIs, I couldn't think of a reasonable way to do so.